proposals/flexible-vectors/README.md

```diff
@@ -1,7 +1,1185 @@
-## Flexible vectors proposal
+# Concrete model for flexible vectors
 
-- Instruction descriptions
-  - [FlexibleVectors.md](FlexibleVectors.md)
-  - [FlexibleVectorsSecondTier.md](FlexibleVectorsSecondTier.md)
-  - [FlexibleVectorsThirdTier.md](FlexibleVectorsThirdTier.md)
+## Terms
 
+- (WASM) compiler: A compiler reads a program source code (eg: C) and translates it into WASM bytecode.
+- target architecture: the architecture where the WASM bytecode is ran.
+- WASM engine: An engine reads WASM bytecode and translates it into native instructions for the architecture it is currently running on, and runs them.
+- element width: size in bits of the element (`float` is 32-bit wide).
+- SIMD width: size in bits of the hardware SIMD registers.
+- SIMD length: number of elements that can fit within a hardware SIMD register (SIMD width / element width).
+- lane: element within a vector
+
+## General goal
+
+The core of the flexible vectors proposal is to a have a single bytecode that can target multiple architectures with different SIMD width.
+This should be as efficient as possible (ie: it should run as fast as possible).
+In order to do that, flexible vectors should abstract the SIMD width away.
+
+List of potential targets:
+
+| ISA           | SIMD width   |
+|:--------------|:-------------|
+| SSE - SSE4.2  | 128          |
+| AVX - AVX2    | 256          |
```

Comment:
avx and avx2 instructions can work on both 128-bit and 256-bit vectors

Reply:
I think the idea was that this is the maximum width

Reply:
ok, fair enough, but at least in upcoming avx10.1 and avx10.2 the maximum vector width (256-bit or 512-bit) will depend on hardware as the 512-bit width is optional in these standards.